scrypto-schema/src/lib.rs

```diff
@@ -184,24 +216,64 @@ impl From<PackageSchema> for IndexedPackageSchema {
 
 impl IndexedBlueprintSchema {
     pub fn num_fields(&self) -> usize {
-        self.tuple_module
-            .as_ref()
-            .map(|(_, fields)| fields.len())
-            .unwrap_or(0)
+        match &self.fields {
+            Some((_, indices)) => indices.len(),
+            _ => 0usize,
+        }
     }
 
-    pub fn field(&self, field_index: u8) -> Option<(u8, LocalTypeIndex)> {
-        self.tuple_module.as_ref().and_then(|(offset, fields)| {
-            let field_index: usize = field_index.into();
-            fields.get(field_index).cloned().map(|f| (*offset, f))
-        })
+    pub fn field(&self, field_index: u8) -> Option<(PartitionOffset, LocalTypeIndex)> {
+        match &self.fields {
+            Some((offset, fields)) => {
+                let field_index: usize = field_index.into();
+                fields
+                    .get(field_index)
+                    .cloned()
+                    .map(|f| (offset.clone(), f))
+            }
+            _ => None,
+        }
     }
 
-    pub fn key_value_store_module_offset(
+    pub fn key_value_store_partition(
+        mut self,
+        collection_index: u8,
+    ) -> Option<(PartitionOffset, ScryptoSchema, BlueprintKeyValueStoreSchema)> {
+        let index = collection_index as usize;
+        if index >= self.collections.len() {
+            return None;
+        }
+
+        match self.collections.swap_remove(index) {
```

Comment:
I'm very surprised by `swap_remove`? Doesn't this change the collection indices, and would break on subsequent calls to `key_value_store_partition`?